lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -41,19 +42,34 @@ export class RedirectHandler extends InteractionHandler {
      * Handle authorization code response in the window.
      * @param hash
      */
-    async handleCodeResponse(locationHash: string): Promise<TokenResponse> {
+    async handleCodeResponse(locationHash: string, browserCrypto?: ICrypto): Promise<TokenResponse> {
         // Check that location hash isn't empty.
         if (StringUtils.isEmpty(locationHash)) {
             throw BrowserAuthError.createEmptyHashError(locationHash);
         }
 
         // Interaction is completed - remove interaction status.
         this.browserStorage.removeItem(BrowserConstants.INTERACTION_STATUS_KEY);
+
+        // Get cached items
+        const requestState = this.browserStorage.getItem(TemporaryCacheKeys.REQUEST_STATE);
+        const cachedNonceKey = this.browserStorage.generateNonceKey(requestState);
+        const cachedNonce = this.browserStorage.getItem(cachedNonceKey);
+        this.authCodeRequest = this.browserStorage.getCachedRequest(requestState, browserCrypto);
+
         // Handle code response.
-        const codeResponse = this.authModule.handleFragmentResponse(locationHash);
+        const authCode = this.authModule.handleFragmentResponse(locationHash, requestState);
+        this.authCodeRequest.code = authCode;
+
         // Hash was processed successfully - remove from cache
         this.browserStorage.removeItem(TemporaryCacheKeys.URL_HASH);
+
+        // Extract user state.
+        const userState = ProtocolUtils.getUserRequestState(requestState);
+
         // Acquire token with retrieved code.
-        return this.authModule.acquireToken(codeResponse);
+        const tokenResponse = await this.authModule.acquireToken(this.authCodeRequest, userState, cachedNonce);
```

Comment:
Leave this for now, but lets find common ground later not to rewrite this for `msal-node`